### ISSUE TITLE
[web] Button becomes un-tappable in Flutter Web when a TextField is present.

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#151292]
An internal customer reported a problem with text fields and buttons (and routes) in the flutter web canvaskit renderer.

<details>
<summary><tt>lib/main.dart</tt></summary>

```dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';

void main() async {
  runApp(
    MyApp(),
  );
}

final goRouter = GoRouter(
  routes: [
    GoRoute(
      path: '/',
      builder: (context, state) => Scaffold(
        body: Center(
          child: TextButton(
            onPressed: () => context.go('/next'),
            child: Text('Next'),
          ),
        ),
      ),
    ),
    GoRoute(
      path: '/next',
      builder: (context, state) => MyHomePage(
        title: 'Next page',
      ),
    ),
  ],
);

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp.router(
      title: 'Flutter Demo',
      routerConfig: goRouter,
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(
          widget.title,
        ),
        automaticallyImplyLeading: false,
        leading: BackButton(
          onPressed: () {
            context.go('/');
          },
        ),
      ),
      body: Column(
        children: [
          TextField(
            autofocus: true,
          ),
        ],
      ),
    );
  }
}
```

</details>

<details>
<summary><tt>web/index.html</tt></summary>

```html
<html>
  <head>
    <script src="flutter.js" defer></script>
  </head>
  <body>
    <script>
      window.addEventListener('load', function (ev) {
        _flutter.loader.loadEntrypoint({
          entrypointUrl: 'main.dart.js',
          onEntrypointLoaded: function(engineInitializer) {
            engineInitializer.initializeEngine().then(function(appRunner) {
              appRunner.runApp();
            });
          }
        });
      });
   </script>
  </body>
</html>
```

</details>

To reproduce:

1. Navigate to and from the nested route using the buttons.
2. Focus, unfocus the text field.
3. Repeat.

Observed behavior:

* Button becomes unresponsive very soon. It happens sooner or later.
* Exceptions:

```
 (errors.dart:296:3)
    at Object.assertFailed (errors.dart:29:3)
    at Object._computeOffsetForInputs (event_position_helper.dart:64:37)
    at Object.computeEventOffsetToTarget (event_position_helper.dart:32:14)
    at [_convertEventsToPointerData] (pointer_binding.dart:1062:30)
    at pointer_binding.dart:1001:9
    at pointer_binding.dart:939:7
    at loggedHandler (pointer_binding.dart:532:9)
    at Object._callDartFunctionFast1 (js_allow_interop_patch.dart:188:27)
    at ret (js_allow_interop_patch.dart:81:15)
```

**Comment by ditman** [comment]
The TL;DR of the issue is that the customer's `index.html` was lacking a `<!DOCTYPE html>` declaration.

Without it, the browser was rendering in ["Quirks" mode](), causing some of our elements (text editing forms) to have a slightly different user-agent stylesheet. In particular, forms have an unexpected `margin-block-end: 1em` CSS that is not present in standards mode.

This caused that the `flt-text-editing-host` (which is supposed to not have a size) to grow by 16px every time a new form was added to the DOM.

We need to:

1. Reset the `margin` CSS for the `form` elements, so text editing doesn't break in quirks mode.
2. Let developers know when they forget to set the doctype on their index.html file, so they can correct it!
